File: ATP 2/Codigos/C e C++/AULA 8/at7_ex5.cpp
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>
#include <utility>

int mdc(int x, int y)
{
    if(x > y)
    {
        int temp = y;
        y = x;
        x = temp;
    }
    if(y%x == 0)
    {
        return x;
    }
    return(mdc(x, y%x));
}

int main()
{
    int a, b, c;
    printf("\nA: ");
    scanf("%d", &a);

    printf("\nB: ");
    scanf("%d", &b);

    printf("\nC: ");
    scanf("%d", &c);

    printf("\nMDC: %d", mdc(mdc(a, b), c));

    getchar();
    return 0;
}

File: Estudos solo/Python/exercicios atividade 7 em python/at7ex5.py

def mdc(a, b):
    if a > b:
        temp = a
        a = b
        b = temp
    if (b%a) == 0:
        return a
    else:
        result = mdc((b%a), a)
        return result
    
x = int(input("X: "))
y = int(input("Y: "))
z = int(input("Z: "))

print(mdc(mdc(x, y), z))